'_WIN32_WINNT': macro redefinition

**Does this issue affect the google-cloud-cpp project?**

This issue affects my project, as settings in the `FindgRPC.cmake` conflict with settings in my project.

**What component of `google-cloud-cpp` is this related to?**

`FindgRPC.cmake`

**Describe the bug**

I cannot compile my project because of the following compiler error:

```txt
CmeSmartStream.cpp : error C2220: the following warning is treated as an error
CmeSmartStream.cpp : warning C4005: '_WIN32_WINNT': macro redefinition
cmake_pch.cxx : message : see previous definition of '_WIN32_WINNT'
CmeSmartStream.cpp : warning C4651: '/D_WIN32_WINNT=0x0A00' specified for precompiled header but not for current compile
```

The reason is because `google-cloud-cpp::pubsub` CMake target defines the macro `_WIN32_WINNT=0x600`, while my project defines `_WIN32_WINNT=0x0A00`. I cannot downgrade `_WIN32_WINNT` to 0x0600, because it is my Company's policy. And there is no workaround/option in the `FindgRPC.cmake` to not define the `_WIN32_WINNT` macro

Following is a snippet from the `FindgRPC.cmake` file:

```cmake
function (_grpc_fix_grpc_target_definitions)
    # Including gRPC headers without this definition results in a build error.
    if (WIN32)
        set_property(
            TARGET gRPC::grpc
            APPEND
            PROPERTY INTERFACE_COMPILE_DEFINITIONS _WIN32_WINNT=0x600)
        set_property(
            TARGET gRPC::grpc++
            APPEND
            PROPERTY INTERFACE_COMPILE_DEFINITIONS _WIN32_WINNT=0x600)
    endif ()
endfunction ()

...

if (gRPC_FOUND)
    _grpc_fix_grpc_cpp_plugin_target()
    _grpc_fix_grpc_target_definitions()
    return()
endif ()
```

Following is how my local project looks like

```cmake
cmake_minimum_required(VERSION 3.10...3.24)

if( WIN32 )
    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_WIN32_WINNT=0x0A00 -W4 -WX")
    set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -D_WIN32_WINNT=0x0A00 -W4 -WX")
endif()

project(google-cloud-cpp-pubsub-quickstart CXX)

find_package(google_cloud_cpp_pubsub REQUIRED)

add_library(my_lib lib.cc)
target_precompile_headers(
    my_lib PUBLIC my_lib_pch.h
)

add_executable(quickstart quickstart.cc)
target_link_libraries(quickstart my_lib google-cloud-cpp::pubsub)

target_precompile_headers(quickstart REUSE_FROM my_lib)
```

**To Reproduce**

1. Follow the instruction to [install the PubSub using vcpkg](https://github.com/googleapis/google-cloud-cpp/tree/main/google/cloud/pubsub/quickstart#using-with-cmake)
2. Download and extract the archive [sources.zip](https://github.com/googleapis/google-cloud-cpp/files/10157758/sources.zip)
3. Open the `make.cmd` and fix the path to vcpkg folder and 
4. Run the `make.cmd`

At the end compiler reports:

```txt
google-cloud-cpp\google\cloud\pubsub\quickstart\bug>cmake --build .build 
MSBuild version 17.4.0+18d5aef85 for .NET Framework
  Checking Build System
  Building Custom Rule C:/Repos/repos/qlt/google-cloud-cpp/google/cloud/pubsub/quickstart/bug/CMakeLists.txt
  cmake_pch.cxx
  lib.cc
  my_lib.vcxproj -> C:\Repos\repos\qlt\google-cloud-cpp\google\cloud\pubsub\quickstart\bug\.build\Debug\my_lib.lib
  Building Custom Rule C:/Repos/repos/qlt/google-cloud-cpp/google/cloud/pubsub/quickstart/bug/CMakeLists.txt
  quickstart.cc
google-cloud-cpp\google\cloud\pubsub\quickstart\bug\quickstart.cc : error C2220: the following warning is treated as an error [C:\Repos\repos\qlt\google-cloud-cpp\google\cloud\pubsub\quickstart\bug\.build\quickstart.vcxproj]
google-cloud-cpp\google\cloud\pubsub\quickstart\bug\quickstart.cc : warning C4005: '_WIN32_WINNT': macro redefinition [C:\Repos\repos\qlt\google-cloud-cpp\google\cloud\pubsub\quickstart\bug\.build\quickstart.vcxproj]
google-cloud-cpp\google\cloud\pubsub\quickstart\bug\.build\CMakeFiles\my_lib.dir\cmake_pch.cxx : message : see previous definition of '_WIN32_WINNT' [C:\Repos\repos\qlt\google-cloud-cpp\google\cloud\pubsub\quickstart\bug\.build\quickstart.vcxproj]
google-cloud-cpp\google\cloud\pubsub\quickstart\bug\quickstart.cc : warning C4651: '/D_WIN32_WINNT=0x0A00' specified for precompiled header but not for current compile [C:\Repos\repos\qlt\google-cloud-cpp\google\cloud\pubsub\quickstart\bug\.build\quickstart.vcxproj]

```

**Expected behavior**

Project is compiled successfully

**Operating system:**

Windows 10

**What compiler and version are you using?**

Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30146 for x64

**What version of `google-cloud-cpp` are you using?**
Please include the output from `git rev-parse HEAD` if you are compiling from source, or the version number from the applicable `google/cloud/*/version.h` file.

```cpp
#define GOOGLE_CLOUD_CPP_VERSION_MAJOR 2
#define GOOGLE_CLOUD_CPP_VERSION_MINOR 4
#define GOOGLE_CLOUD_CPP_VERSION_PATCH 0
```

**Additional context**
Add any other context about the problem here.

--- a/cmake/FindgRPC.cmake
+++ b/cmake/FindgRPC.cmake
@@ -102,22 +102,6 @@ function (_grpc_fix_grpc_cpp_plugin_target)
     endif ()
 endfunction ()
 
-# The gRPC::* targets sometimes lack the right definitions to compile cleanly on
-# WIN32
-function (_grpc_fix_grpc_target_definitions)
-    # Including gRPC headers without this definition results in a build error.
-    if (WIN32)
-        set_property(
-            TARGET gRPC::grpc
-            APPEND
-            PROPERTY INTERFACE_COMPILE_DEFINITIONS _WIN32_WINNT=0x600)
-        set_property(
-            TARGET gRPC::grpc++
-            APPEND
-            PROPERTY INTERFACE_COMPILE_DEFINITIONS _WIN32_WINNT=0x600)
-    endif ()
-endfunction ()
-
 # First try to use the `gRPCConfig.cmake` or `grpc-config.cmake` file if it was
 # installed. This is common on systems (or package managers) where gRPC was
 # compiled and installed with `CMake`.
@@ -131,7 +115,6 @@ endif ()
 
 if (gRPC_FOUND)
     _grpc_fix_grpc_cpp_plugin_target()
-    _grpc_fix_grpc_target_definitions()
     return()
 endif ()
 

